app/decorators/plant_trial_submission_decorator.rb

```diff
@@ -43,7 +43,7 @@ def plant_population
 
   def trait_descriptors
     tds = object.content.step02.trait_descriptor_list.try(:select, &:present?) || []
-    numeric_tds = tds.select { |td| td.to_s.match /\A\d+\z/ }
+    numeric_tds = tds.select { |td| td.to_s.match(/\A\d+\z/) }
```

Comment:
I believe that params passed from for are always seen as strings and it
stays like that when they are stored in submission's content.

2016-06-16 19:20 GMT+02:00 Tomasz Gubała <notifications@github.com>:

> In app/decorators/plant_trial_submission_decorator.rb
> <https://github.com/eSpectrum-IT/brassica/pull/546#discussion_r67386923>:
>
> > @@ -43,7 +43,7 @@ def plant_population
> >
> >    def trait_descriptors
> >      tds = object.content.step02.trait_descriptor_list.try(:select, &:present?) || []
> > -    numeric_tds = tds.select { |td| td.to_s.match /\A\d+\z/ }
> > +    numeric_tds = tds.select { |td| td.to_s.match(/\A\d+\z/) }
>
> Yes, it seems we have old_ in two places and existing_ in two others. So
> please change to numeric to existing and also try to replace the old_
> occurrence in submission.rb (the other one is in specs, you may ignore
> that one).
>
> Currently one is not able to call a new TD with a number, since I closed
> the vocabulary (it's only Trait.name values now). However, I can easily
> imagine us allowing a free text version, since the TO ontology is rather
> limiting for our users. Therefore I asked about that. Are existing tds
> stored in the array as integers?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eSpectrum-IT/brassica/pull/546/files/6e425ba70db7c5a52ebf2ed70395ee5b74d3bf9c#r67386923>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAA2ySvGxpI6DzfbLL0FiK456QvZfIo3ks5qMYXrgaJpZM4I2hPI>
> .
>



-- 
Tomasz Szymczyszyn